hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -143,31 +141,21 @@ public CryptoCreateRecordBuilder newRecordBuilder() {
 
     /* ----------- Helper Methods ----------- */
 
-    /**
-     * Validate the basic fields in the transaction body that does not involve checking with dynamic
-     * properties or state. This check is done as part of the pre-handle workflow.
-     * @param op the transaction body
-     * @return OK if the transaction body is valid, otherwise return the appropriate error code
-     */
-    private ResponseCodeEnum pureChecks(@NonNull final CryptoCreateTransactionBody op) {
-        if (op.initialBalance() < 0L) {
-            return INVALID_INITIAL_BALANCE;
-        }
-        if (!op.hasAutoRenewPeriod()) {
-            return INVALID_RENEWAL_PERIOD;
-        }
-        if (op.sendRecordThreshold() < 0L) {
-            return INVALID_SEND_RECORD_THRESHOLD; // FUTURE: should this return
-            // SEND_RECORD_THRESHOLD_FIELD_IS_DEPRECATED
-        }
-        if (op.receiveRecordThreshold() < 0L) {
-            return INVALID_RECEIVE_RECORD_THRESHOLD; // FUTURE: should this return
-            // RECEIVE_RECORD_THRESHOLD_FIELD_IS_DEPRECATED
-        }
-        if (op.hasProxyAccountID() && !op.proxyAccountID().equals(AccountID.DEFAULT)) {
-            return PROXY_ACCOUNT_ID_FIELD_IS_DEPRECATED;
-        }
-        return OK;
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        final var op = txn.cryptoCreateAccountOrThrow();
+        validateTruePreCheck(op.initialBalance() >= 0L, INVALID_INITIAL_BALANCE);
```

Comment:
The team should probably discuss use of these "validateZZZ" methods.  I find them significantly less clear than the simple conditional, and they add method call overhead just to hide a single statement.
Not worth blocking this PR, but definitely something to be discussed more broadly.

Reply:
Possibly in a subgroup, but not in any venue where I recall hearing it.